force-app/main/default/classes/DMLOperations.cls

```diff
@@ -146,11 +160,43 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+        Account account;
+
+        // find the Account by name; if not found, create a new one
+        try {
+            account = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        } 
+        catch (System.QueryException e) {
+            account = new Account(Name = accountName);
+            upsert account; 
+        }
+    
+        // list of Opportunities for upsert
+        List<Opportunity> opportunitiesToUpsert = new List<Opportunity>();
     
+        // Loop through each Opportunity name and check for existing Opportunities
+        for (String oppName : oppNames) {
+            try {
+                Opportunity existingOpportunity = 
```

Comment:
Looks like you're using existingOpportunity to do a query but the variable is not used anywhere for anything. Try to use this variable to figure out if opportunities exists in the org already.